### ISSUE TITLE
[2.0.0-RC6] ZStream *Par methods may adopt .withParallelism syntax

### ISSUE THREAD

**Issue by ak0rz** [zio/zio#6774]
Currently methods *Par* in ZStream requires explicit parallelism as the first argument, which was [discouraged in ZIO](https://zio.dev/next/howto/migrate/zio-2.x-migration-guide#compositional-concurrency).

Maybe we should consider make the same change for ZStream.

**Comment by adamgfraser** [comment]
Yes. I think we need to improve some of the infrastructure in streams to really support this but will think about what we do for this for ZIO 2.0.
